tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -158,8 +153,13 @@ void read_from_pcie(volatile tt_l1_ptr uint32_t *& prefetch_q_rd_ptr,
         prefetch_q_rd_ptr = (volatile tt_l1_ptr uint32_t*)prefetch_q_base;
     }
 
-    // NOT_STALLED Persists. STALL_NEXT -> STALLED.
-    stall_state = static_cast<StallState>(stall_state << 1);
+    // If prefetcher must stall after this fetch, wait for data to come back, and move to next state.
```

Comment:
I don't like this version as much. I think the "stall" concept is a "get cmds" concept and "read_from_pcie just does reads

Reply:
Sounds good, was mainly put here to prevent duplication, will move to right after read_from_pcie() call.

Reply:
I see, was mainly to reduce needing to duplicate the same code twice by caller, will stick in inline function and call twice.